### PR TITLE
remove logstash/pipeline

### LINKED ISSUE #11236: remove ruby execution
With java execution we no longer need the ruby one. There are know issues blocking this removal:

- [ ] https://github.com/elastic/logstash/issues/11175

_The Java Execution Engine has been the default engine since Logstash 7.0, and is capable of using plugins written in either Ruby or Java; removing the Ruby Execution Engine will not affect our ability to run existing pipelines, but will greatly reduce the surface area of all features and fixes within the engine(s), since we will no longer need to target two engines simultaneously._

### PR DESCRIPTION
**Pull request by kaisecheng**
Fixed: https://github.com/elastic/logstash/issues/11236

This PR replace remove ruby pipeline

### CODE REVIEW

**Comment by kaisecheng** [comment]
jenkins test this please

File: ci/unit/docker-setup.sh [hunk 1]
```diff
@@ -1,55 +1,59 @@
 #!/usr/bin/env bash
 
- # This is intended to be run the plugin's root directory. `ci/unit/docker-test.sh`
+# This is intended to be run the plugin's root directory. `ci/unit/docker-test.sh`
 # Ensure you have Docker installed locally and set the ELASTIC_STACK_VERSION environment variable.
 set -e
 
- VERSION_URL="https://raw.githubusercontent.com/elastic/logstash/master/ci/logstash_releases.json"
+pull_docker_snapshot() {
```
Comment (andsel) [inline ci/unit/docker-setup.sh]: This seems quite similar to https://github.com/logstash-plugins/.ci/blob/e824dc9bb5db58d04d86773ae30ec90d4a84626b/docker-setup.sh#L7-L12 there is any way to share it?

**Review by andsel — COMMENTED** [review]
I've left a doubt, on sharing common docker config with `.ci` project

**Review by andsel — APPROVED** [review]
LGTM

**Pull request merged**